opal/mca/btl/tcp/btl_tcp.c

```diff
@@ -531,3 +533,64 @@ void mca_btl_tcp_dump(struct mca_btl_base_module_t* base_btl,
     }
 #endif /* OPAL_ENABLE_DEBUG && WANT_PEER_DUMP */
 }
+
+
+/*
+ * A blocking recv on a non-blocking socket. Used to receive the small
+ * amount of connection information that identifies the endpoints
+ * endpoint.
+ */
+
```

Comment:
Note: we are setting SO_RCVTIMEO only in mca_btl_tcp_component_recv_handler where it is blocking call. so we are handling both cases. But I feel, I need to change the function comments and add more details to make it more clear (may be change function name here)